### ISSUE TITLE
Duplicate edges with `--ts-config --ts-pre-compilation-deps`

### ISSUE THREAD

**Issue by jfirebaugh** [sverweij/dependency-cruiser#111]
### Expected Behavior
In the dot output, between any pair of vertices A and B there is at most one directed edge from A to B.

### Current Behavior
When using the `--ts-config --ts-pre-compilation-deps` CLI options and a `tsconfig.json` with `"module": "commonjs"`, depcruise sometimes creates two outward edges to the same vertex.

#### tsconfig.json
```
{
  "compilerOptions": {
    "module": "commonjs"
  }
}
```

#### scratch/a.ts
```
import { foo } from './b';
foo();
```

#### scratch/b.ts
```
export function foo() {}
```

Now run `depcruise --ts-config --ts-pre-compilation-deps -T dot scratch`. Observe that the output contains:

```
    "scratch/a.ts" -> "scratch/b.ts"
    "scratch/a.ts" -> "scratch/b.ts"
```

### Your Environment

* Version used: 4.14.0
* Node version: v10.14.1
* TypeScript version: 3.3.1

**Comment by sverweij** [comment]
hi @jfirebaugh - thanks - that's indeed unintuitive. I've fixed it at least visually (see #113), by making the dot output a strict digraph (part of tomorrow's release 🤞). More structural solution.

#### Why does this happen?
By default dependency-cruiser compiles everything to javascript and infers dependencies from the javascript AST. The typescript pre-compilation dependencies it gets from the typescript AST. Most pre-compilation dependencies will also be in the javascript AST. Typically dependency-cruiser de-duplicates them because both would be es6 dependencies. In this issue, however, typescript compiler compiles down to commonjs, making it two dependencies - an es6 one from the typescript AST, and a commonjs one from the javascript one. 

#### Fixes 
It's sort-of logical this happens, but also extremely unintuitive, so there will be PR's ahead to fix it.

- Low hanging fruit: make the dot output address the visual bit (done; PR #113)
- Go all out on the typescript AST and also get the commonjs dependencies from there (and stop using the compiled javascript for typescript dependency). Might make the extraction process faster.
- de-duplicate the dependency-graph by giving precedence to es6 module inclusions (exclusively in ts settings, or also for other code bases).

#### Workarounds
Here's two ways to work around it until the release hits npm:
- Specify `--module-systems es6`  as an extra command line parameter - which will work fine only if there's no commonjs module dependencies in your typescript _xor_ ...
- use a separate tsconfig that inherits from your original `tsconfig`, but specifies ESNEXT as module system (`"module": "ESNEXT"`) and pass that to dependency-cruiser with `--ts-config tsconfig.depcruise.json`

**Comment by sverweij** [comment]
b.t.w. - if you want to already try out the visual fix - it's on npm in the beta channel as `dependency-cruiser@4.15.0-beta-2`

**Comment by sverweij** [comment]
update: I've shipped the visual fix shipped with 4.15.0 just now. 

The more structural internal (#114) fix is almost done - I know it works from unit & manual tests, but I still need to set up a mini-integration test - so that will be in a next release. 

If you want to use the more structural fix earlier: just holler - I'll just push a beta to npm.

**Comment by jfirebaugh** [comment]
@sverweij Thanks so much for the detailed response and quick fix. I can tell you are putting a lot of time and effort into dependency-cruiser. It's amazingly useful!

**Comment by sverweij** [comment]
thanks @jfirebaugh 😊 

(I've merged #114  into develop, so it'll be part of the next release and the duplicates don't show up anymore in the other reporters)

**Comment by sverweij** [comment]
With the merge & release of #119 this issue is also solved for other reporters.
